src/Generators/Generator.php

```diff
@@ -312,98 +312,45 @@ public function getTemplatePath()
+    protected function renderTemplate(string $template, array $params): string
     {
-        $class = $this->$attribute;
+        $renderer = function (...$args) {
+            extract($args[1], EXTR_OVERWRITE);
+            require $args[0];
+        };
+
+        $obInitialLevel = ob_get_level();
+        ob_start();
+        ob_implicit_flush(0);
         try {
-            if (class_exists($class)) {
-                if (isset($params['extends'])) {
-                    if (ltrim($class, '\\') !== ltrim($params['extends'], '\\') && !is_subclass_of($class, $params['extends'])) {
-                        $this->addError($attribute, "'$class' must extend from {$params['extends']} or its child class.");
-                    }
+            $renderer->bindTo($this)($template, $params);
+            return ob_get_clean();
+        } catch (\Throwable $e) {
+            while (ob_get_level() > $obInitialLevel) {
+                if (!@ob_end_clean()) {
+                    ob_clean();
                 }
-            } else {
-                $this->addError($attribute, "Class '$class' does not exist or has syntax error.");
-            }
-        } catch (\Exception $e) {
-            $this->addError($attribute, "Class '$class' does not exist or has syntax error.");
-        }
-    }
-
-    /**
-     * An inline validator that checks if the attribute value refers to a valid namespaced class name.
-     * The validator will check if the directory containing the new class file exist or not.
-     * @param string $attribute the attribute being validated
-     * @param array $params the validation options
-     */
-    public function validateNewClass($attribute, $params)
-    {
-        $class = ltrim($this->$attribute, '\\');
-        if (($pos = strrpos($class, '\\')) === false) {
-            $this->addError($attribute, "The class name must contain fully qualified namespace name.");
-        } else {
-            $ns = substr($class, 0, $pos);
-            $path = Yii::getAlias('@' . str_replace('\\', '/', $ns), false);
-            if ($path === false) {
-                $this->addError($attribute, "The class namespace is invalid: $ns");
-            } elseif (!is_dir($path)) {
-                $this->addError($attribute, "Please make sure the directory containing this class exists: $path");
             }
-        }
-    }
-
-    /**
-     * Checks if message category is not empty when I18N is enabled.
-     */
-    public function validateMessageCategory()
-    {
-        if ($this->enableI18N && empty($this->messageCategory)) {
-            $this->addError('messageCategory', "Message Category cannot be blank.");
+            throw $e;
         }
     }
 
     /**
      * @param string $value the attribute to be validated
      * @return bool whether the value is a reserved PHP keyword.
      */
-    public function isReservedKeyword($value)
+    public function isReservedKeyword($value): bool
```

Comment:
Can we make `Validator` from this method?

Reply:
Let's do it in the next PRs